### PR TITLE
Feedback to Finni

### PR DESCRIPTION
**Pull request by cbourdage**
Hey man, 

Went through this guy first and will try to get to the other one. Instead of trying to leave feedback in some weird way google doc or email, I figured I would just open a PR against master so you can see the changes for the specific feedback.

Feel free to ask questions via the PR review process or respond in the PR directly in anyway you see fit. Also, note, I did NOT test this I purely just read through the code and provided some insight and feedback.

Hope this helps.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -8,12 +8,23 @@ import LocationForm from './components/LocationForm';
...
 
+    constructor(props) {
+        super(props);
+
+        // @rfinni you can define your state here, you can also do it like on line 11 above
         this.state = {
             weatherConditions: { loaded: false }
         }
+
+        // ideally, you want to handle function binding here so when you use or pass functions down
```
Comment (rfinni) [inline src/App.js:24]: Nice! I hadn't thought of that.

File: src/App.js [hunk 2]
```diff
@@ -8,12 +8,23 @@ import LocationForm from './components/LocationForm';
 
 class App extends Component {
 
-    constructor() {
-        super();
+    state = {
```
Comment (rfinni) [inline src/App.js:11]: @cbourdage is there a difference either way?

File: src/components/CardFooter.js [hunk 3]
```diff
@@ -21,6 +35,15 @@ export default class CardFooter extends Component {
     }
 
     render() {
+        // A neat trick to variable setting is to "deconstruct" your props object. If you are using
```
Comment (rfinni) [inline src/components/CardFooter.js:38]: @cbourdage this is awesome.

File: src/components/CardSummary.js [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
-                <div className="left short-info">
-                    <span className="temp">{Math.floor(this.props.data.temp)}&deg;</span>
-                    <p className="weather-text">{this.props.data.weather}</p>
-                </div>
-                <div className={"icon-" + this.props.data.icon + " right weather-block"}></div>
-            </div>
-        );
-    }
-}
+// Same note here about the "functional" component. Also, I didn't mention this in
```
Comment (rfinni) [inline src/components/CardSummary.js:3]: @cbourdage ah yeah looks like I missed that here, good call.

File: src/components/LocationForm.js [hunk 5]
```diff
@@ -1,28 +1,52 @@
...
 
+// This looks good. A couple of notes
+//  1. React is deprecating the use of refs as a string, so instead, use their function callback
+//  2. bind submit in the constructor
+//  3. always declare any props so any engineer knows what a component needs in order to work
 export default class LocationForm extends Component {
 
-    submit(e) {
-        e.preventDefault();
+  static defaultProps = {
```
Comment (rfinni) [inline src/components/LocationForm.js:9]: @cbourdage what's the difference between this and defining propTypes?

File: src/components/WeatherCard.js [hunk 6]
```diff
@@ -3,28 +3,36 @@ import React, { Component, PropTypes } from 'react';
...
-                {this.props.data.loaded ?
-                    <div className="tile-inner">
-                        <header className="city">
-                            <h1>{this.props.data.city}</h1>
-                        </header>
-                        <CardSummary data={this.props.data} />
-                        <CardFooter data={this.props.data} reloadView={this.props.reloadView} />
-                    </div>
-                    : <div className="tile-inner-loading"><p>Fetching weather data</p></div> }
+const WeatherCard = ({ data, reloadView }) => (
```
Comment (rfinni) [inline src/components/WeatherCard.js:15]: @cbourdage is it usually good practice to do this if render() is the only method within a component?

File: src/App.js (hunk 2 above)
Comment (cbourdage) [reply src/App.js:11]: There isn't. This approach is more preferred but there are times when you need to do it in the constructor.

File: src/components/LocationForm.js (hunk 5 above)
Comment (cbourdage) [reply src/components/LocationForm.js:9]: So this was a typo mistake. This should be `static propTypes` but there is also a secondary `static defaultProps` that allows you to define default values for our properties, especially for default values.

File: src/components/WeatherCard.js (hunk 6 above)
Comment (cbourdage) [reply src/components/WeatherCard.js:15]: Yep! Anytime it's just a render component, just make it functional. This means react no longer needs to run any diffs on state and it also doesn't need to process the life-cycle methods. It just checks if the leaf is different and then renders it.

**Pull request closed without merging**